### ISSUE TITLE
Async implementation

### ISSUE THREAD

**Issue by elpiel** [dbrgn/shtcx-rs#32]
I would like to ask if there is interest in making an async version of the crate?

I'm using the sensor in an embassy project and while some other sensors use async version of the I2C bus and I can use [embassy_embedded_hal::shared_bus::asynch::i2c::I2cDevice](https://docs.embassy.dev/embassy-embedded-hal/git/default/shared_bus/asynch/i2c/struct.I2cDevice.html) to share the bus across multiple tasks, it's not possible to use a driver crate with a blocking I2C impl.

**Comment by dbrgn** [comment]
So far I haven't yet played with embassy. I assume it uses the traits from [embedded-hal-async](https://docs.rs/embedded-hal-async/1.0.0-rc.1/embedded_hal_async/i2c/index.html), right?

**Comment by elpiel** [comment]
Yes, indeed. There's already a RC version for -async so it might get stabilised especially with AFIT (async fn. In traits) hitting 1.75

**Comment by elpiel** [comment]
@dbrgn both `embedded-hal` and `embedded-hal-async` are on v1.0.0
It would be amazing to update the crate and support the stable version of the hal crates.

**Comment by dbrgn** [comment]
Support for e-h 1.0 is tracked in #31. Once that's in, it probably makes sense to add an optional feature for `embedded-hal-async` as well.

**Comment by elpiel** [comment]
Can you expect an async version soon?

**Comment by dbrgn** [comment]
> Can you expect an async version soon?

Only if somebody steps up and contributes an initial implementation.

Something that should be taken into account is that #38 should probably be handled before adding async support (since it might have interactions, potentially even benefits).

**Comment by Fristi** [comment]
I've started a bit on this. See: https://github.com/dbrgn/shtcx-rs/compare/main...Fristi:shtcx-rs:feature/async-support

Not sure if you would like the module exports like this, it will be a breaking change

Didn't update the docs and such, only have runtime code compiling (asynchronous + blocking)

**Comment by SergioGasquez** [comment]
Hi @Fristi! Any updates on your work adding async support? Thanks for sharing it, it would be awesome if we could upstream it!
